models/phonebook.py

```diff
@@ -0,0 +1,95 @@
+from models.contact import Contact
+
+
+class Contacts():
+    """Contains and manages all contacts. This is a phonebook instance.
+
+    Args:
+        This class takes no arguments
+
+    Returns:
+        Returns nothing.
+
+    """
+
+    def __init__(self):
+        self.data = {}
+
+    def add(self, firstname, surname, number):
+        """Function for adding contacts to the phonebook instance.
+
+        Args:
+            firstname: First name of the contact owner.
+            surname: Surname of the contact owner.
+            number: The actual phone number of the owner.
+
+        Returns:
+            Returns "Contact added successfully!" when a contact is added
+            with no issues.
+            Returns an error message, "Contact exists!" when an attempt to duplicate is noted.
+
+        """
+
+        # only add non-existent contacts
+        if number not in self.data.keys():
+            self.contact = Contact(firstname, surname, number)
+            self.data[self.contact.number] = [firstname, surname]
+            return("Contact added successfully!")
```

Comment:
What do you think about adding a conditional statement to prove that the contact record was added successfully? This will enable you to avoid logical errors.

Reply:
Like an IF statement to ascertain the existence of the record?

Reply:
Yeah